lib/gui/pages/main/controllers/image-selection.js

```diff
@@ -122,20 +123,22 @@ module.exports = function(
    * ImageSelectionController.selectImageByPath('path/to/image.img');
    */
   this.selectImageByPath = (imagePath) => {
-    imageStream.getImageMetadata(imagePath)
-      .then(this.selectImage)
-      .catch((error) => {
-        const imageError = errors.createUserError({
-          title: 'Error opening image',
-          description: messages.error.openImage({
-            imageBasename: path.basename(imagePath),
-            errorMessage: error.message
-          })
+    $timeout(() => {
```

Comment:
As far as I can see, the `$timeout` is only needed for `this.selectImage`. If so, can we do

```js
imageStream.getImageMetadata(imagePath)
    .then((imageMetadata) => {
        $timeout(() => {
            this.selectImage(imageMetadata);
        });
    });
```

?